### PR TITLE
Support for Redis Sentinel 

### PR DESCRIPTION
**Pull request by purinda**
WIP: Checking connection status is only added for master instances.

SentinelMonitor class is responsible for communicating with the sentinel
server in regard to finding out master and slave instances of redis
running.

Modified RedisDriver to figure out running redis instances from Sentinel
service.

### CODE REVIEW

**Comment by purinda** [comment]
Shouldn't, but about to find out. Deploying DB configuration and UATs

File: src/Bravo3/Orm/Drivers/Redis/RedisDriver.php [hunk 1]
```diff
@@ -34,6 +34,11 @@ class RedisDriver implements DriverInterface
     protected $client;
 
     /**
+     * @var SentinelMonitor
+     */
+    private $sentinel = null;
```
Comment (jordonsc) [inline src/Bravo3/Orm/Drivers/Redis/RedisDriver.php]: Why private, everything else is protected?

File: src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+     * Create a new Sentinel Monitor.
+     *
+     * If more than one sentinel server parameters are provided, the first
+     * sentinel that connects will be used for retrival of information.
+     *
+     * @param mixed $params
+     */
+    public function __construct($connection_params = null)
+    {
+        foreach($connection_params as $connection) {
```
Comment (jordonsc) [inline src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php]: Missing a space after foreach

File: src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * Create a new Sentinel Monitor.
+     *
+     * If more than one sentinel server parameters are provided, the first
+     * sentinel that connects will be used for retrival of information.
+     *
+     * @param mixed $params
+     */
+    public function __construct($connection_params = null)
```
Comment (jordonsc) [inline src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php:22]: `$connection_params` must be an array, type-hint it and update the docblock

File: src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    /**
+     * Return an array of SENTINEL servers in the form that can be
+     * passed to Predis Client.
+     *
+     * @param  string $master_name
+     * @return array
+     */
+    public function findSentinels($master_name = 'mymaster')
+    {
+        $sentinels = $this->client->sentinel('sentinels', $master_name);
```
Comment (jordonsc) [inline src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php]: `return $this->client->sentinel('sentinels', $master_name);`

File: src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * Return an array of MASTER servers in the form that can be
+     * passed to Predis Client.
+     *
+     * @return array
+     */
+    public function findMasters()
+    {
+        $masters = $this->client->sentinel('masters');
```
Comment (jordonsc) [inline src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php]: `return $this->getConnectionParams($this->client->sentinel('masters'), 'master');`

File: src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+     * @return array
+     */
+    public function findSlaves($master_name = null)
+    {
+        $slaves = [];
+
+        if (null !== $master_name) {
+            $slaves = $this->client->sentinel('slaves', $master_name);
+        } else {
+
```
Comment (jordonsc) [inline src/Bravo3/Orm/Drivers/Redis/SentinelMonitor.php]: Space shouldn't be here

File: src/Bravo3/Orm/Drivers/Redis/RedisDriver.php (hunk 1 above)
Comment (purinda) [inline src/Bravo3/Orm/Drivers/Redis/RedisDriver.php]: for no reason tbh. I will use protected.

**Comment by jordonsc** [comment]
:+1:

**Pull request merged**